File: ls8.c
#include <stdio.h>
#include <stdlib.h>
#include "cpu.h"
/**
 * Main
 */
int main(int argc, char **argv)
{
    struct cpu cpu;
    if (argc != 2)
    {
        fprintf(stderr, "usage: main <filename>\n");
        exit(1);
    }

    char *filename = argv[1];

    cpu_init(&cpu);
    cpu_load(&cpu, filename);
    cpu_run(&cpu);

    return 0;
}

File: ls8.py
argv_err_msg = """
Error: You must specify a file name as a command line argument.
"""

import sys
from cpu import *

if len(sys.argv) != 2:
    print(argv_err_msg)
    sys.exit(1)
else:
    file_name = sys.argv[1]

cpu = CPU()
cpu.load(file_name)
cpu.run()